modules/rosa-create-objects.adoc

```diff
@@ -273,6 +273,81 @@ Create a cluster with autoscaling enabled on the default worker machine pool.
 $ rosa create cluster --cluster-name=mycluster -region=us-east-1 --enable-autoscaling --min-replicas=2 --max-replicas=5
 ----
 
+[id="rosa-create-external-auth-provider_{context}"]
+== create external-auth-provider
+
+Add an external authentication provider instead of an internal OIDC provider to approve users to access your cluster.
+
+.Syntax
+[source,terminal]
+----
+$ rosa create external-auth-provider --cluster=<cluster_name> | <cluster_id> [arguments]
+----
+
+.Arguments
+[cols="30,70"]
+|===
+|Option |Definition
+
+|--claim-mapping-groups-claim <string>     
+|Required. Describes rules on how to transform information from an ID token into a cluster identity.
+
+|--claim-validation-rule <strings>
+|ClaimValidationRules are rules that are applied to validate token claims to authenticate users. The input will be in a <claim>:<required_value> format. To have multiple claim validation rules, you could separate the values by ','. The input could be in a <claim>:<required_value>,<claim>:<required_value> format.
+
+|--claim-mapping-username-claim <string>
+|The name of the claim that should be used to construct user names for the cluster identity.
+
+a|--cluster <cluster_name>\|<cluster_id>
+|Required. The name or ID of the cluster to which the IDP will be added.
+
+|--console-client-id <string>
+|The identifier of the OIDC client from the OIDC provider.
+
+|--console-client-secret <string>          
+|The value of the client secret that is associated with your console app registration. If you do not include the client secret, this command uses a public OIDC OAuthClient.
```

Comment:
> associated with your console app registration

This is Entra ID-specific. Use OAuth2/OIDC lingo instead.
> If you do not include the client secret, this command uses a public OIDC OAuthClient.

That statement is false. The web console requires client secret.

Reply:
I don't quite know how to resolve this text.

@lucasponce can I get some assistance?

Reply:
As commented by @yingzhanredhat per https://github.com/openshift/openshift-docs/pull/73839/files#r1574341120. I do see in some documents that we refer this to be console app registration. @stlaz Is OAuth2/OIDC a more appropriate way to be documented?

For the 2nd part, I think @stlaz is correct. Web console requires client secret & token, but web console is not required to create an external auth for OIDC, so maybe we can rephrase it a bit.